sdk/servicebus/azure-servicebus/tests/stress_tests/test_stress_queues.py

```diff
@@ -111,6 +111,46 @@ def test_stress_queue_receive_and_delete(self, servicebus_namespace_connection_s
                                        receivers = [sb_client.get_queue_receiver(servicebus_queue.name, mode=ReceiveSettleMode.ReceiveAndDelete)],
                                        duration=timedelta(seconds=60))
 
+        result = stress_test.Run()
+        assert(result.total_sent > 0)
+        assert(result.total_received > 0)
+
+
+    @pytest.mark.liveTest
+    @pytest.mark.live_test_only
+    @CachedResourceGroupPreparer(name_prefix='servicebustest')
+    @ServiceBusNamespacePreparer(name_prefix='servicebustest')
+    @ServiceBusQueuePreparer(name_prefix='servicebustest')
+    def test_stress_queue_unsettled_messages(self, servicebus_namespace_connection_string, servicebus_queue):
+        sb_client = ServiceBusClient.from_connection_string(
+            servicebus_namespace_connection_string, debug=False)
+
+        stress_test = StressTestRunner(senders = [sb_client.get_queue_sender(servicebus_queue.name)],
+                                       receivers = [sb_client.get_queue_receiver(servicebus_queue.name)],
+                                       duration = timedelta(seconds=350),
+                                       should_complete_messages = False)
+
+        result = stress_test.Run()
+        # This test is prompted by reports of an issue where enough unsettled messages saturate a service-side cache
+        # and prevent further receipt.
+        assert(result.total_sent > 2500)
+        assert(result.total_received > 2500)
```

Comment:
emmm, I saw the comment but still didn't what's the real goal here.

It this the case that we meet service-side's storage limitation -- service side can no longer accept new messages?
I know there's a queue setting called Max Queue Size In GB which is 1 GB by default.
Are you testing the 1GB limitation here?

Reply:
I unfortunately don't know if the symptoms I heard of interact with Max Queue Size, but I assume not since it was # of message based, and was more to do with that SB had some sort of service-side cache such that enough throughput without settlement caused receipt to halt?  This was a discussion that one of the other languages had run into some time ago, and I had made a note to ensure we covered this scenario in our stress-tests, if only as a regression check.

Reply:
I'm can OK this as it does no harm. 

I'm assuming that the issue doesn't take place in our SDK, right? but would be interested in how other languages meet up with the issue.

and the throttling value -- 2500 messages, is it an empirical value?

Reply:
Yes, it was an explicit threshold; and correct, it was a service-side issue.  If we wanted to be defensive, knowing that there was this pattern of failure, it'd probably be germane to up the test threshold to like, 3x/4x.  I can follow up with the other language reps, since at this point my memory of this is a todo list item and vague recollections of the conversation that spawned it.